### PR TITLE
Fix client options flow to Http line in SDKs.

### LINKED ISSUE #36296: [BUG] ProxyOptions passed in ClientSecretCredentialBuilder.clientOptions() has no effect
**Describe the bug**
I want to use `BlobServiceClient` with Service Principal authentication from behind a forward proxy.
I use `ClientSecretCredentialBuilder` to build the SP credential.
If I pass the `ProxyOptions` in `ClientSecretCredentialBuilder.clientOptions()`, it has no effect and it cannot access `login.microsoftonline.com` via my proxy.
`ManagedIdentityCredentialBuilder` is also affected.

It may be a similar issue as #28900 / #28783.

There exists a workaround: use `NettyAsyncHttpClientBuilder.proxy()` and then pass the built netty client in `ClientSecretCredentialBuilder.httpClient()`.
But the simpler API call (without building a new `HttpClient`) should work too.

***Code Snippet***
```
        ProxyOptions proxyOptions = new ProxyOptions(...);
        
        ClientSecretCredential credential = new ClientSecretCredentialBuilder()
                .tenantId(...)
                .clientId(...)
                .clientSecret(...)
                .clientOptions(new HttpClientOptions().setProxyOptions(proxyOptions))
                .build();
```

**Expected behavior**
`ProxyOptions` passed in `ClientSecretCredentialBuilder.clientOptions()` should work.

### PR DESCRIPTION
**Pull request by g2vinay**
Fixes Client Options Wiring to Http Pipeline in following SDKs:

Resolves #36296 

- Identity
- KeyVault Secrets, Keys, Certificates, Administration
- App Config
- Tables

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/PowershellManager.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ private Mono<Boolean> readData(BufferedReader reader, StringBuilder powerShellOu
     private Mono<Boolean> canRead(BufferedReader reader) {
         Supplier<Boolean> supplier = () -> {
             int pause = 62;
-            int maxPause = Platform.isMac() ? this.waitPause : 500;
```
Comment (joshfree) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/PowershellManager.java:125]: could you only include http proxy changes in this PR and also associate the PR with the customer-reported github issue for better tracking?

**Review by joshfree — APPROVED** [review]
Looks good - except for the unrelated powershellcredential changes and not being linked to the customer-reported issue.

**Review by billwert — APPROVED** [review]
LGTM

**Review by vcolin7 — CHANGES_REQUESTED** [review]
The changes for Key Vault look good, but we're missing a couple of builders: `KeyVaultBackupClientBuilder` & `KeyVaultSettingsClientBuilder`.

**Review by vcolin7 — APPROVED** [review]

**Pull request merged**